### PR TITLE
Created the Users::BankAccountsController

### PR DESCRIPTION
**Pull request by igorrfc**
Some needed changes to accomplish this task:

- [x] Create the Attachment model to manage the bank account's authorization documents;

- [x] Update the BankAccount adding an association with the authorization documents;

- [x] Create the Users::BankAccountsController;

### CODE REVIEW

File: app/controllers/users/bank_accounts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+class Users::BankAccountsController < ApplicationController
+  after_filter :verify_authorized, except: [ :index ]
```
Comment (marclerodrigues) [inline app/controllers/users/bank_accounts_controller.rb]: When passing only one action you can omit the brackets.
```ruby
 after_action :verify_authorized, expect: :index
```

File: app/controllers/users/bank_accounts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  before_action :has_list_permission?, only: [ :index ]
```
Comment (marclerodrigues) [inline app/controllers/users/bank_accounts_controller.rb]: The same thing here

File: app/controllers/users/bank_accounts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  def index
+    head :ok
+  end
+
+  def new
+    @bank_account = BankAccount.new(user: current_user)
+    authorize @bank_account
+
+    render nothing: true
```
Comment (marclerodrigues) [inline app/controllers/users/bank_accounts_controller.rb]: At the index action you're using `head :ok` and here you're using `render nothing: true`. Why the difference?. Both do the same and `head...` is preferred over `render nothing...`.

File: app/controllers/users/bank_accounts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    @bank_account.attributes = bank_account_params
+
+    if @bank_account.save
+      flash[:notice] = t(:success, scope: 'user.bank_account.create')
+    else
+      flash[:alert] = @bank_account.errors.full_messages.to_sentence
+    end
+
+    render nothing: true
```
Comment (andrezacm) [inline app/controllers/users/bank_accounts_controller.rb]: mesma coisa que o marclerodrigues falou aqui sobre o render nothing

**Pull request merged**